icu4c/source/common/dictbe.cpp

```diff
@@ -199,13 +199,13 @@ ThaiBreakEngine::ThaiBreakEngine(DictionaryMatcher *adoptDictionary, UErrorCode
 {
     UTRACE_ENTRY(UTRACE_UBRK_CREATE_BREAK_ENGINE);
     UTRACE_DATA1(UTRACE_INFO, "dictbe=%s", "Thai");
-    fThaiWordSet.applyPattern(UNICODE_STRING_SIMPLE("[[:Thai:]&[:LineBreak=SA:]]"), status);
+    UnicodeSet thaiWordSet(UNICODE_STRING_SIMPLE("[[:Thai:]&[:LineBreak=SA:]]"), status);
```

Comment:
(Optional) We could get rid of using the archaic UNICODE_STRING_SIMPLE macro now that we can use Unicode string literals. Use this instead: 
```UnicodeString(u"[[:Thai:]&[:LineBreak=SA:]]")```

Even more efficient, but more awkward, would be to build the set using properties directly, avoiding the need to parse the pattern string. Probably not worth the trouble and loss of easy readability.

Same comments apply to all the other changes.

Reply:
Yes, please change these string literals to using `u"..."` directly.
But keep the pattern strings. The parser is not *that* expensive, compared with computing the underlying sets and the intersection.